cmd/gateway-gcs.go

```diff
@@ -294,8 +282,48 @@ func (l *gcsGateway) ListBuckets() ([]BucketInfo, error) {
 	return b, nil
 }
 
-// DeleteBucket delete a bucket on GCS
+// DeleteBucket delete a bucket on GCS.
 func (l *gcsGateway) DeleteBucket(bucket string) error {
+	itObject := l.client.Bucket(bucket).Objects(l.ctx, &storage.Query{Delimiter: slashSeparator, Versions: false})
+	// We list the bucket and if we find any objects we return BucketNotEmpty error. If we
+	// find only "minio.sys.temp/" then we remove it before deleting the bucket.
+	gcsMinioPathFound := false
+	nonGCSMinioPathFound := false
+	for {
```

Comment:
@balamurugana the protocol we have decided to follow is:

"never complicate the code for the sake of unit tests, instead write simple readable code and make tests work for it. If it is a stateless function then it should be unit tested, if it is not stateless then functional test should be used to test it."

Another protocol we need to follow of when to create a new function:
* when a piece of code needs to be reusable
* when the logic of a function is complex and becomes difficult to read/maintain, it should be broken into smaller functions that can be used to compose the function that we need.

Hence we should not separate this for loop just for unit testing. We should however cover testing this function in functional testing.